rfd/0034-desktop-access-windows.md

```diff
@@ -0,0 +1,268 @@
+exposure to Kerberos tickets (even when using AAD as SSO provider) and we can't
+assume that all clients use Windows machines, Teleport will not use TGTs for
+RDP authentication.
+
+### Host discovery
+
+When a user starts a new desktop session, they must specify the Windows host to
+connect to. Internally, Teleport tracks known Windows hosts using
+`WindowsDesktop` objects.
+
+There are 3 ways that `windows_desktop_service` discovers Windows hosts to
+register:
+- hardcoded list of standalone hosts provided in the config file (see
+  [configuration](#configuration))
+- list of Active Directory-enrolled hosts obtained from AD via LDAPS (LDAP over
+  SSL)
+  - LDAP library: https://pkg.go.dev/github.com/go-ldap/ldap/v3
+- local host, when running on a Windows machine in agent mode
+
+### Concurrent sessions
+
+RDP supports concurrent user sessions on the same host. It allocates a virtual
+desktop for each user to isolate their activities. However, it only allows a
+single session per user per host. If a user starts a new session on the host
+that they are already logged into, they will log out the other session. This is
+RDP behavior we cannot change.
+
+### Session recording
+
+RDP video output uses bitmaps and not a standard video encoding format.
+To allow playback of session recordings, we have 2 options:
+- record bitmap updates with timestamps and replay them using JS on a canvas.
+- encode finished bitmap recording into MP4 or WebM
+
+The latter allows recording export and sharing, and can be played back in many
+existing applications. However, video encoding is a complex and
+resource-intensive process that has to be done after the session is complete,
+because video container formats must encode the total duration upfront.
+
+Teleport will record active desktop sessions into the simplest possible format
+and then transcode them to MP4 or WebM asynchronously. Transcoding can be done
+using `ffmpeg` if we don't find a solid encoder in Go or Rust.
+
+### Configuration
+
+New `teleport.yaml` section for `windows_desktop_service`:
+
+```yaml
+windows_desktop_service:
+  enabled: yes # default false
+  listen_addr: 0.0.0.0:3028
+  public_addrs: [rdp.example.com:3028]
+  mode: "gateway" # or "agent"
+  # (optional) ldap contains hostname and credentials for the LDAP server on
+  # the Active Directory domain controller.
+  # If specified, windows hosts will be automatically discovered by Teleport.
+  #
+  # Note: Teleport will only connect to LDAP over TLS and will always
+  # validate the server certificate.
+  ldap:
+    host: ldap.example.com
+    username: "ldap_user"
+    password: "ldap_pass"
+    # optional CA cert to use for LDAP server certificate validation.
+    ca_cert: "/var/lib/teleport/ldap_ca.pem"
+  # (optional) static_hosts is a list of hostnames to register as
+  # WindowsDesktop objects in Teleport.
+  # These are usually non-AD hosts.
+  static_hosts:
```

Comment:
may be just hosts? Also take a look at Roman's RFD. Lots of times users would want to use terraform provider to specify hosts, but AD based discovery is the most important part, I agree

Reply:
renamed to `hosts`
and yes, dynamic registration will also be supported